util/storage-samples/storage-owner-sample/src/main/java/org/fidoalliance/fdo/storage/OwnerSysModule.java

```diff
@@ -61,11 +67,31 @@ public Composite getState() {
 
   @Override
   public void setServiceInfo(Composite kvPair, boolean isMore) {
-
+    String key = kvPair.getAsString(Const.FIRST_KEY);
+    Object value = kvPair.get(Const.SECOND_KEY);
+
+    switch (key) {
+      case FdoSys.KEY_ACTIVE:
```

Comment:
If active=false, then:
1. Should state's 4th key be set to false explicitly, along with the 5th (false) and 6th keys (true), since there's nothing to be sent?
2. Since the 2nd key is set to 0 here, the conditions at methods 'isMore()' and 'nextMessage()', that is condition 'if (resIndex < 0)' won't be triggered. Should the 2nd key be set to -1, or the condition be changed to 'if (resIndex <= 0)' ?

Reply:
yes fixing point 1)  but regarding point 2) resIndex <0 is a special condition that indicates that the module is not ready to respond (e.g the devmod as not finished).  resIndex >=0 means the module is ready to respond but may or may not be done.   IsMore() always returns done=true and ismore=false if resIndex >= resList.Length.  else if (resIndex < IndexLength we still have messages to process.   I did add a resList.clear() when active=false.  This will always trigger checkMessage to return done=true and isMore=false for a future requests.